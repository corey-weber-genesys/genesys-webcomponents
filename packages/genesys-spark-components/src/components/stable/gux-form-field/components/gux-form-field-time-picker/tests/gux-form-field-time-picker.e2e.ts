import { E2EPage, newE2EPage } from '@stencil/core/testing';
import {
  newSparkE2EPage,
  a11yCheck
} from '../../../../../../test/e2eTestUtils';

const axeExclusions = [];

async function newNonrandomE2EPage({
  html
}: {
  html: string;
}): Promise<E2EPage> {
  const page = await newE2EPage();

  await page.evaluateOnNewDocument(() => {
    Math.random = () => 0.5;
  });
  await page.setContent(html);
  await page.waitForChanges();

  return page;
}

describe('gux-form-field-time-picker', () => {
  describe('#render', () => {
    describe('label-position', () => {
      [
        '',
        'label-position="above"',
        'label-position="beside"',
        'label-position="screenreader"'
      ].forEach((componentAttribute, index) => {
        const html = `
        <gux-form-field-time-picker ${componentAttribute}>
          <gux-time-picker value="07:00"></gux-time-picker>
          <label slot="label">Label</label>
        </gux-form-field-time-picker>
        `;

        it(`should render component as expected (${index + 1})`, async () => {
          const page = await newNonrandomE2EPage({ html });
          const element = await page.find('gux-form-field-time-picker');
          const elementShadowDom = await element.find(
            'pierce/.gux-form-field-container'
          );

          expect(element.outerHTML).toMatchSnapshot();
          expect(elementShadowDom).toMatchSnapshot();
        });

        it(`should be accessible (${index + 1})`, async () => {
          const page = await newSparkE2EPage({ html });

          await a11yCheck(page, axeExclusions);
        });
      });
    });

    describe('interval', () => {
      ['interval="15"', 'interval="30"', 'interval="60"'].forEach(
        (componentAttribute, index) => {
          const html = `
          <gux-form-field-time-picker>
            <gux-time-picker value="07:00" ${componentAttribute}></gux-time-picker>
            <label slot="label">Label</label>
          </gux-form-field-time-picker>
        `;

          it(`should render component as expected (${index + 1})`, async () => {
            const page = await newNonrandomE2EPage({ html });
            const element = await page.find('gux-form-field-time-picker');
            const elementShadowDom = await element.find(
              'pierce/.gux-form-field-container'
            );

            expect(element.outerHTML).toMatchSnapshot();
            expect(elementShadowDom).toMatchSnapshot();
          });

          it(`should be accessible (${index + 1})`, async () => {
            const page = await newSparkE2EPage({ html });

            await a11yCheck(page, axeExclusions);
          });
        }
      );
    });

    describe('help', () => {
      const html = `
      <gux-form-field-time-picker>
      <gux-time-picker value="09:00"></gux-time-picker>
      <label slot="label">Select Time</label>
      <span slot="help">This is a help message</span>
    </gux-form-field-time-picker>
      `;

      it('should render component as expected', async () => {
        const page = await newNonrandomE2EPage({ html });
        const element = await page.find('gux-form-field-time-picker');
        const elementShadowDom = await element.find(
          'pierce/.gux-form-field-container'
        );

        expect(element.outerHTML).toMatchSnapshot();
        expect(elementShadowDom).toMatchSnapshot();
      });

      it('should be accessible', async () => {
        const page = await newSparkE2EPage({ html });

        await a11yCheck(page, axeExclusions);
      });
    });
  });
});
