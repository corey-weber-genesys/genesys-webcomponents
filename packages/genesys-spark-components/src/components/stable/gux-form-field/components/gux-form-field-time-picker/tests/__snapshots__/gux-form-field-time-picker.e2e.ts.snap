// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-form-field-time-picker #render help should render component as expected 1`] = `"<gux-form-field-time-picker hydrated=\\"\\"> <gux-time-picker value=\\"09:00\\" id=\\"gux-form-field-input-i\\" aria-describedby=\\"gux-form-field-help-i\\" hydrated=\\"\\"></gux-time-picker> <label slot=\\"label\\" for=\\"gux-form-field-input-i\\">Select Time</label> <span slot=\\"help\\" id=\\"gux-form-field-help-i\\">This is a help message</span> </gux-form-field-time-picker>"`;

exports[`gux-form-field-time-picker #render help should render component as expected 2`] = `null`;

exports[`gux-form-field-time-picker #render interval should render component as expected (1) 1`] = `"<gux-form-field-time-picker hydrated=\\"\\"> <gux-time-picker value=\\"07:00\\" interval=\\"15\\" id=\\"gux-form-field-input-i\\" hydrated=\\"\\"></gux-time-picker> <label slot=\\"label\\" for=\\"gux-form-field-input-i\\">Label</label> </gux-form-field-time-picker>"`;

exports[`gux-form-field-time-picker #render interval should render component as expected (1) 2`] = `null`;

exports[`gux-form-field-time-picker #render interval should render component as expected (2) 1`] = `"<gux-form-field-time-picker hydrated=\\"\\"> <gux-time-picker value=\\"07:00\\" interval=\\"30\\" id=\\"gux-form-field-input-i\\" hydrated=\\"\\"></gux-time-picker> <label slot=\\"label\\" for=\\"gux-form-field-input-i\\">Label</label> </gux-form-field-time-picker>"`;

exports[`gux-form-field-time-picker #render interval should render component as expected (2) 2`] = `null`;

exports[`gux-form-field-time-picker #render interval should render component as expected (3) 1`] = `"<gux-form-field-time-picker hydrated=\\"\\"> <gux-time-picker value=\\"07:00\\" interval=\\"60\\" id=\\"gux-form-field-input-i\\" hydrated=\\"\\"></gux-time-picker> <label slot=\\"label\\" for=\\"gux-form-field-input-i\\">Label</label> </gux-form-field-time-picker>"`;

exports[`gux-form-field-time-picker #render interval should render component as expected (3) 2`] = `null`;

exports[`gux-form-field-time-picker #render label-position should render component as expected (1) 1`] = `"<gux-form-field-time-picker hydrated=\\"\\"> <gux-time-picker value=\\"07:00\\" id=\\"gux-form-field-input-i\\" hydrated=\\"\\"></gux-time-picker> <label slot=\\"label\\" for=\\"gux-form-field-input-i\\">Label</label> </gux-form-field-time-picker>"`;

exports[`gux-form-field-time-picker #render label-position should render component as expected (1) 2`] = `null`;

exports[`gux-form-field-time-picker #render label-position should render component as expected (2) 1`] = `"<gux-form-field-time-picker label-position=\\"above\\" hydrated=\\"\\"> <gux-time-picker value=\\"07:00\\" id=\\"gux-form-field-input-i\\" hydrated=\\"\\"></gux-time-picker> <label slot=\\"label\\" for=\\"gux-form-field-input-i\\">Label</label> </gux-form-field-time-picker>"`;

exports[`gux-form-field-time-picker #render label-position should render component as expected (2) 2`] = `null`;

exports[`gux-form-field-time-picker #render label-position should render component as expected (3) 1`] = `"<gux-form-field-time-picker label-position=\\"beside\\" hydrated=\\"\\"> <gux-time-picker value=\\"07:00\\" id=\\"gux-form-field-input-i\\" hydrated=\\"\\"></gux-time-picker> <label slot=\\"label\\" for=\\"gux-form-field-input-i\\">Label</label> </gux-form-field-time-picker>"`;

exports[`gux-form-field-time-picker #render label-position should render component as expected (3) 2`] = `null`;

exports[`gux-form-field-time-picker #render label-position should render component as expected (4) 1`] = `"<gux-form-field-time-picker label-position=\\"screenreader\\" hydrated=\\"\\"> <gux-time-picker value=\\"07:00\\" id=\\"gux-form-field-input-i\\" hydrated=\\"\\"></gux-time-picker> <label slot=\\"label\\" for=\\"gux-form-field-input-i\\">Label</label> </gux-form-field-time-picker>"`;

exports[`gux-form-field-time-picker #render label-position should render component as expected (4) 2`] = `null`;
